src/components/select/option/option.component.js

```diff
@@ -3,9 +3,17 @@ import PropTypes from "prop-types";
 import StyledOption from "./option.style";
 
 const Option = React.forwardRef(
-  ({ text, children, onSelect, value, isHighlighted, hidden }, ref) => {
+  (
+    { text, children, onSelect, value, isHighlighted, hidden, onClick },
+    ref
+  ) => {
     function handleClick() {
-      onSelect({ text, value });
+      if (!onClick) {
+        onSelect({ text, value });
+      } else {
+        onSelect();
```

Comment:
Why does it need to call `onSelect` with no parameters here?

Reply:
When I added `onClick` to the `Option` component I interfered in Select behavior. I have to call onSelect to let `Select component` know that we made an action